### PR TITLE
Fixed concurrency issue in configuration file usage

### PR DESCRIPTION
**Pull request by sweco-sebhar**
Cherry picked from https://github.com/mapfish/mapfish-print/pull/329.

### CODE REVIEW

File: src/main/java/org/mapfish/print/MapPrinter.java [hunk 1]
```diff
@@ -202,12 +204,18 @@ public void printClientConfig(JSONWriter json) throws JSONException {
      * Stop the thread pool or others.
      */
     @PreDestroy
-    public void stop() {
-        if (config != null) {
+    public synchronized void stop() {
+    	usages--;
+        if (config != null && usages <= 0) {
+        	usages = 0;
```
Comment (sbrunner) [inline src/main/java/org/mapfish/print/MapPrinter.java]: Indentation issue ..., in general it's better to don't use tabs !

**Comment by sbrunner** [comment]
What do you think @tsauerwein? For me that's looks good :-)

**Comment by tsauerwein** [comment]
I don't know MFP v2 at all, but I am trying to understand the problem: There is a `MapPrinter` instance for each config file, and a `MapPrinter` instance could be used to process multiple requests in parallel. Now, the issue was that the config file of a `MapPrinter` was closed before all requests have finished. Is this right?

If so, the changes seem to fix the issue. But there are several places where the indentation is wrong.

**Comment by sweco-sebhar** [comment]
Sorry I don't understand the details of this patch only that I'm told it solves issues regarding concurrent request. I'll fix the indentation if you want to accept the patch regardless of limited knowledge about it. :)

**Comment by tsauerwein** [comment]
Thanks! Maybe @mbarto can give some details?

**Comment by sweco-sebhar** [comment]
Yeah hopefully, I don't know much about it except I've been told it helps but that the original PR was submitted to the wrong branch.

**Comment by mbarto** [comment]
Hi, tsauerwein analysis of patch behaviour is correct. The purpose is to avoid closing a MapPrinter when it's in use by many parallel threads.

**Comment by tsauerwein** [comment]
Thanks for the clarification @mbarto! I think we can merge this PR.

@mbarto, if you are interested in having this fix also for branch `2.1.x`, could you also please fix the indentation like @sweco-sebhar did in commit https://github.com/sweco/mapfish-print/commit/80a8f0e62de1ff93678d76d9b19dc27e86a6d85b.

**Comment by mbarto** [comment]
Hi, I updated #329 to fix indentation issues, can you please check if it's ok now for merging?

**Pull request merged**